### ISSUE TITLE
Analysis page: Bad performance

### ISSUE THREAD

**Issue by Ein-Tim** [corona-warn-app/cwa-website#1806]
## Where to find the issue
- https://www.coronawarn.app/de/analysis/
- https://www.coronawarn.app/en/analysis/

## Describe the issue
The analysis page is very unresponsive and slow. 
Here's a video: 

https://user-images.githubusercontent.com/67682506/135767003-2d5e380f-e9ca-4f1f-a2df-b020a7769dce.mp4

**Comment by dsarkar** [comment]
@Ein-Tim Thanks for reporting. @FabioFuschi-SNK FYI. Any proposals how to approach this issue? Thanks.

**Comment by MikeMcC399** [comment]
@Ein-Tim 
For performance issues it's good to have an actual measurement, like you click on a certain button and it took a measured amount of time to produce a result.
If you have the current value, then you should specify the maximum expected value that you would consider acceptable.

Finally it would be good to know what system you were using and what the network parameters were.

---
I timed what I saw you doing in the video on a Windows 10 system with Google Chrome and a measured download internet connection speed of 38.8 Mbps.

On the page https://www.coronawarn.app/de/analysis/ I clicked on "Gesamter Zeitraum" with "Täglich" selected and it took 2.2 seconds to re-display all graphs. I had to set the zoom to 50% so I could see all graphs at once.

I found that speed acceptable, but of course your measurements and expectations may be different.

**Comment by MikeMcC399** [comment]
@Ein-Tim 
Using browser developer tools it seems that the action you were taken is just redrawing the graphs without using any network connection, so that will probably depend on the processing power of your system. It will be interesting to see the response from @FabioFuschi-SNK in any case.

**Comment by Ein-Tim** [comment]
@MikeMcC399 

> For performance issues it's good to have an actual measurement, like you click on a certain button and it took a measured amount of time to produce a result.

I agree with you, but as you can see, in the end of the video, the site just does not produce a result at all anymore. 

> Finally it would be good to know what system you were using and what the network parameters were.

You're right! As always, I did this testing with my Mac Book Air 2020 (M1) and used Safari. I also tested under Edge (Chromium based), same result. I was connected to the internet via 5 GHZ WLAN with:
- Ping: 19 ms
- Download: 108.66 Mbps
- Upload: 43.02 Mbps

---

> Using browser developer tools it seems that the action you were taken is just redrawing the graphs without using any network connection, so that will probably depend on the processing power of your system. It will be interesting to see the response from @FabioFuschi-SNK in any case.

True, I just used the developer tools myself and yes, indeed there is no more downloading of any numbers going on once on the page. 
I recorded another video, this time with the "Aktivitätsanzeige" open on the other half of the display (the quality is really bad as GitHub limits file uploads to 10MB, sorry. Below is a zip were you can find the uncompressed video):

https://user-images.githubusercontent.com/67682506/135857046-ba616fb0-2687-4d71-a5ba-901f5acf3501.MP4

As you can see, coronawarn.app is sometimes responsible for nearly all of the processing done by the CPU. I'm very sure that there can be done something to enhance this, viewing the page on older devices must be very slow currently! 

---

Zip with the video:
[Slow.mov.zip](https://github.com/corona-warn-app/cwa-website/files/7278210/Slow.mov.zip)

**Comment by FabioFuschi-SNK** [comment]
The bad performance comes from the apexchart DOM rendering task. Especially Safari is on this task really slow, i measured time up to 15s on my Safari. The rendering time increases with the data points. The longer the period, the longer the graphs will take to render. That is not a bug, it is the normal behavior if the page.

**Comment by Ein-Tim** [comment]
@FabioFuschi-SNK 

Ah, okay! Thanks for the explanation! Not only Safari has this problem, but Microsoft Edge (Chromium based) is also very slow.

Is there any alternative or any improvement possible here? 
Because, sad to say, but the normal user will not wait more than ca. 10 sec. for the graphs to show up, if it just shows loading the normal user assumes somethings broken and leaves the page. 😅

**Comment by jucktnich** [comment]
@Ein-Tim you could show an explaining text after some secs

**Comment by Ein-Tim** [comment]
@jucktnich 

Would be one option, however, normal users will still not wait for more than ca. 15 sec. (I assume).
In general it's really sad that this is the intended behavior, if it's already slow on my MacBook Air with a M1 chip, mid- or low-end devices must be very slow on displaying the pages content...

**Comment by jucktnich** [comment]
Wouldn't it be an idea to create svg pictures to show instead for a locally generated graph? (Or even only as a placeholder?)

**Comment by Ein-Tim** [comment]
@FabioFuschi-SNK 

> The bad performance comes from the apexchart DOM rendering task.

Is there an alternative available? Because, e.g. I don't have any performance problems viewing [WHOs corona dashboard](https://covid19.who.int/).

**Comment by jucktnich** [comment]
Btw for me (iOS 15.1 Public Beta 2; iPad Air 4; A14 Bionic; SFSafariViewController (parent process GitHub)) it's pretty fast. Weird, maybe there was an improvement with WebKit 15 (and macs don't got the new update yet)?
